test/common.py

```diff
@@ -174,6 +192,31 @@ def run_train_translate(train_params: str,
                 lines_equiv = f.readlines()
             assert all(a == b for a, b in zip(lines, lines_equiv))
 
+        # Test restrict-lexicon
+        if restrict_lexicon:
+            # fast_align lex table
+            lex_path = os.path.join(work_dir, "lex")
+            generate_fast_align_lex(lex_path)
+            # Top-K JSON
+            json_path = os.path.join(work_dir, "json")
+            params = "{} {}".format(sockeye.lexicon.__file__,
+                                    _LEXICON_PARAMS_COMMON.format(input=lex_path,
+                                                                  model=model_path,
+                                                                  json=json_path))
+            with patch.object(sys, "argv", params.split()):
+                sockeye.lexicon.main()
+            # Translate corpus with restrict-lexicon
+            out_path = os.path.join(work_dir, "out-restrict.txt")
+            params = "{} {} {} {}".format(sockeye.translate.__file__,
+                                          _TRANSLATE_PARAMS_COMMON.format(model=model_path,
+                                                                          input=dev_source_path,
+                                                                          output=out_path),
+                                          translate_params,
+                                          _TRANSLATE_PARAMS_RESTRICT.format(json=json_path))
+            with patch.object(sys, "argv", params.split()):
+                sockeye.translate.main()
```

Comment:
should we also compute BLEU on the output of translate with lexicon restriction and return it from this function? We could then explicitly test in the integration tests for not loosing too much BLEU when using it.

Reply:
That would be a good system test.  It looks like we're getting a BLEU score of about 0.001 on the fast integration tests so unfortunately there's not much room to test for degradation.

Reply:
Yes, true. I actually meant the system tests that run for longer and produce reasonable ppls and BLEU scores. I always confuse the two...